### PR TITLE
Add layer activations to summaries gteq level 3

### PR DESCRIPTION
**Pull request by Islandman93**
For dense, conv2d, lstm layers

### CODE REVIEW

**Comment by michaelschaarschmidt** [comment]
Thanks! Just for me to keep track, what's the current rational for all the different summary levels?

**Comment by Islandman93** [comment]
It was
0: Loss (always created so no need to check summary_level)
1: Output activations, Qvalues or action logits (still need to do this in distributions)
2: Trainable Variables
3: Intermediate layer activations

Though I don't really like levels anymore. A better system IMO is more like a list of the summaries you want like ["loss", "trainable_variables", "output_activations", "intermediate_activations"] but that's a PR for a different time

**Comment by michaelschaarschmidt** [comment]
Ok yeah, there is probably some room to discuss this but can merge this now
